haralyzer/assets.py

```diff
@@ -371,6 +381,35 @@ def get_total_size(self, entries):
                 size += entry['response']['bodySize']
         return size
 
+    def get_total_size_content(self, entries):
+        """
+        Returns the total size of a collection of entries - uncompressed.
+
+        NOTE: use with har file generated with chrome-har-capturer
+        NOTE: not used at the moment
```

Comment:
Why is this not used at the moment? In other words, do the current methods for getting total content size currently working for chrome-har-capturer documents?

Reply:
The value of `bodySize` in HAR files generated with chrome-har-capturer is "-1". I don't know why and what exactly this means.

When using chrome-har-capturer, if one want to get the full size (not transferred) should use `entry['response']['content']['size']`. However the functions to use this are not implemented.  Maybe will be better to delete it for know.

Reply:
Got it, let me look into some options for us.

Reply:
I think I'm wrong here and this should be deleted. The problem looks to appear in chrome-har-capturer on http/2 hosts. Also my solution is not correct - the value of `bodySize` and the value of `entry['response']['content']['size']` are not the same thing. I should delete this.

Here is the [link](https://github.com/cyrus-and/chrome-har-capturer/issues/27#issuecomment-237573281).

Reply:
Hmmm, so it sounds like the core of the issue then relates to the data that can be acquired in HAR files for HTTP 2.0 specifically?

Reply:
I'm not sure if Mozilla have the same issue, but with Chrome, yes.

Reply:
Got it. So, eventually we are going to need to add some checks at a higher level perhaps. Checking for certain keys and falling back to others if they do not exist. It would make sense that HAR files from HTTP 2.0 might have different data. For your purposes, you just want to be able to get the transferred sizes if they exist (using the new methods you added), correct?

Reply:
Yes, that is correct. I need the transferred sizes of elements.

It must be noted, the new methods are available only with HAR files, generated with Chrome. I'm using the `_transferSize` (not part of HAR spec) provided by Chrome, not implemented in other browsers.

Reply:
Sounds good, I am okay with merging it as is then, since it does not affect any current use cases for the tool. I think the better way to handle this (up to you if you want to implement it or let me do it), is to raise an exception if a method gets called on an unsupported HAR file (i.e. - UnsupportedHARFormat error or something like that). This goes not just for your use case, but any time somebody tries to access a key that does not exist, as the HAR specification is bound to change as well as be different based on the provider creating the HAR file.

Reply:
Great, let's keep this function and do the merge. Would be great if you add the exceptions. :)

Reply:
Cool, I will try to finish reviewing, merge and release by EOD tomorrow.

Reply:
Sorry, @max0xff I can remember where we landed on the `get_total_size_content` method. Sounds like it is not used and not needed? If this is the case, would you mind pulling it back out?

Reply:
Hi @mrname, yes it's not used and not needed. I've removed the method. :)